docs/reference/terminology.md

```diff
@@ -1,29 +1,947 @@
+
+**Dropless MoE:**
+MoE without capacity limits that processes all routed tokens, avoiding token dropping at the cost of variable computation.
+
+**Sparse MoE:**
+Only selected experts are activated per token, not all experts.
+
+**Megablox:**
+Efficient MoE implementation using grouped matrix multiplication (GMM) for handling ragged batches of tokens routed to different experts.
+
+**GMM (Grouped Matrix Multiplication):**
+Pallas kernel for efficient sparse/irregular grouped GEMMs in MoE, handling different batch sizes per expert.
+
+**Expert Capacity:**
+Maximum number of tokens each expert can process in a given batch.
+
+**EP (Expert Parallelism):**
+Sharding strategy that distributes experts across devices.
+
+**Routing Groups:**
+Grouping mechanism for routing in large MoE models to improve load balancing.
+
+---
+
+## Quantization
+
+**Quantization:**
+Reducing numerical precision from FP32/BF16 to lower-bit formats like INT8/FP8 to reduce memory and increase speed.
+
+**QAT (Quantization-Aware Training):**
+Training with simulated quantization effects, inserting fake quantization operations into the forward pass.
+
+**PTQ (Post-Training Quantization):**
+Quantizing weights and activations after training completion.
+
+**AQT (Accurate Quantized Training):**
+Google's quantization library for JAX, providing tools for QAT.
+
+**Qwix:**
+Recommended successor to AQT with a non-intrusive API for quantization in JAX.
+
+**INT8:**
+8-bit integer quantization format, reducing memory by 2-4x compared to FP32/BF16.
+
+**FP8:**
+8-bit floating-point format for quantization, maintaining better dynamic range than INT8.
+
+**BF16 (Brain Float 16):**
+16-bit floating point format with same exponent range as FP32 but reduced mantissa, offering a good balance between precision and memory.
+
+**FP32:**
+32-bit floating point, full precision format.
+
+**Dynamic Range Quantization:**
+Quantization with per-batch or per-tensor statistics for scaling factors.
+
+**Static Quantization:**
+Fixed quantization parameters determined during calibration.
+
+**STE (Straight-Through Estimator):**
+Gradient approximation technique for non-differentiable quantization operations in backward pass.
+
+**Calibration:**
+Process of determining optimal quantization scaling factors using representative data.
+
+---
+
+## Memory Optimization
+
+**Gradient Accumulation:**
+Splitting batch into micro-batches and accumulating gradients before updating parameters, enabling larger effective batch sizes with limited memory.
+
+**Gradient Accumulation Steps:**
+Number of micro-batches to accumulate before performing a parameter update.
+
+**Rematerialization (Remat) / Activation Checkpointing:**
+Recomputing activations in backward pass instead of storing them, trading compute for memory.
+
+**Remat Policy:**
+Strategy for which activations to save vs recompute (options: `full`, `minimal`, `minimal_flash`, `minimal_offloaded_flash`, `save_dot_except_mlp`, `save_qkv_proj`, etc.).
```

Comment:
`minimal_flash` will be deprecated soon, could you update them based on: https://github.com/AI-Hypercomputer/maxtext/blob/9204d6bbbf8bb19a05ebed72a55cfec687e0e044/src/MaxText/configs/base.yml#L282C34-L282C54